public/app/percona/shared/components/Elements/PlatformConnectedLoader/PlatformConnectedLoader.tsx

```diff
@@ -1,28 +1,23 @@
-import React, { FC, useCallback } from 'react';
-import { StoreState } from 'app/types';
+import React, { FC } from 'react';
 import { Messages } from './PlatformConnectedLoader.messages';
-import { PermissionLoader } from '../PermissionLoader';
 import { useSelector } from 'react-redux';
-import { getPerconaUser } from 'app/percona/shared/core/selectors';
-import { EmptyBlock } from '../EmptyBlock';
+import { getPerconaSettings, getPerconaUser } from 'app/percona/shared/core/selectors';
 import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
 
 export const PlatformConnectedLoader: FC = ({ children }) => {
-  const featureSelector = useCallback((state: StoreState) => !!state.perconaSettings.isConnectedToPortal, []);
+  const { isConnectedToPortal } = useSelector(getPerconaSettings);
   const { isPlatformUser } = useSelector(getPerconaUser);
 
-  const checkForPlatformUser = useCallback(() => {
-    if (isPlatformUser) {
-      return children;
+  if (isPlatformUser) {
+    return <>{children}</>;
+  } else {
+    if (isConnectedToPortal) {
+      return <ErrorMessage dataTestId="not-platform-user" message={Messages.platformUser} />;
     }
-    return <EmptyBlock dataTestId="not-platform-user">{Messages.platformUser}</EmptyBlock>;
-  }, [isPlatformUser, children]);
+    if (!isConnectedToPortal) {
```

Comment:
But then everything not defined will end up with "Not connected to Portal" and nothing will end up with "Insufficient access permissions." is that expected behavior? I thought that we talked about that other use cases should end up with "Insufficient access permissions." if there are any. But you are right that it should be isConnectedToPortal === false since this is javascript.

Reply:
Added isAuthorized flag for unauthorized usecases